src/parser/lexer.l

```diff
@@ -13,7 +13,7 @@ int yycolumn = 1;
 
 #define YY_USER_ACTION yycolumn += yyleng; \
     tok.pos = yycolumn; \
-    tok.s = strdup(yytext);
```

Comment:
Why do you think we can work with the original string?

Reply:
I _think_ we can do this because of the way that C stores string literals in static memory, but I can't say definitively that this is a safe or portable practice.

This change resolved a memory leak and did not generate any Valgrind warnings or unexpected results while I tested it.